EventListener/Doctrine/UploadListener.php

```diff
@@ -2,61 +2,43 @@
 
 namespace Vich\UploaderBundle\EventListener\Doctrine;
 
-use Doctrine\Common\EventArgs;
+use Doctrine\ORM\Event\PreFlushEventArgs;
+use Doctrine\ORM\Events;
 
 /**
  * UploadListener
  *
  * Handles file uploads.
  *
+ * @author Konstantin Myakshin <koc-dp@yandex.ru>
  * @author Kévin Gomez <contact@kevingomez.fr>
  */
 class UploadListener extends BaseListener
 {
     /**
-     * The events the listener is subscribed to.
-     *
-     * @return array The array of events.
+     * {@inheritdoc}
      */
     public function getSubscribedEvents()
     {
-        return array(
-            'prePersist',
-            'preUpdate',
-        );
+        return array(Events::preFlush);
```

Comment:
this breaks for people using the ODM rather than the ORM as they won't have the constant defined